lib/options-handler.ts

```diff
@@ -105,6 +105,7 @@ export type ClientOptionsType = {
     doCache: boolean;
     thirdPartyIntegrationEnabled: boolean;
     statusTrackingEnabled: boolean;
+    compilerVersionsUrl: string;
```

Comment:
If I assign it the type string | null, it is no longer a CacheableValue as described in types/cache.interfaces.ts
That type is used to create a hash from the options

It's probably fine to add string|null but I'm not super familiar with this bit of code...
Making it `compilerVersionsUrl?: string;` might make more sense, and that doesn't need changes to the interface

But I don't know if there's a reason you suggested null instead of undefined